packages/bolt-connection/src/bolt/request-message.js

```diff
@@ -286,7 +286,7 @@ function buildTxMetadata (bookmark, txConfig, database, mode, impersonatedUser)
   if (!bookmark.isEmpty()) {
     metadata.bookmarks = bookmark.values()
   }
-  if (txConfig.timeout) {
+  if (txConfig.timeout != null) {
```

Comment:
It's a good practice use '===' and '!==' instead of '==' and '!=' for comparisons in Javascript. The regular comparison methods try to  do type conversion during the comparison and it could lead to strange behaviours.

Reply:
Since we'd like the same behaviour for undefined, issue solved.